src/model/layer/cudnn_rnn.cc

```diff
@@ -0,0 +1,306 @@
+    dimA[0] = batchsize;
+    dimA[1] = inputSize;
+    dimA[2] = 1;
+    strideA[0] = dimA[2] * dimA[1];
+    strideA[1] = dimA[2];
+    strideA[2] = 1;
+    CUDNN_CHECK(cudnnSetTensorNdDescriptor(x_descs_[i], GetCudnnDataType(dtype), 3,
+                                         dimA, strideA));
+    dimA[0] = batchsize;
+    dimA[1] = hiddenSize_ * numDirections;
+    dimA[2] = 1;
+    strideA[0] = dimA[2] * dimA[1];
+    strideA[1] = dimA[2];
+    strideA[2] = 1;
+    CUDNN_CHECK(cudnnSetTensorNdDescriptor(y_descs_[i], GetCudnnDataType(dtype), 3,
+                                         dimA, strideA));
+  }
+  
+  dimA[0] = numLayers_;
+  dimA[1] = batchsize;
+  dimA[2] = hiddenSize_ * numDirections;
+  strideA[0] = dimA[2] * dimA[1];
+  strideA[1] = dimA[2];
+  strideA[2] = 1;
+  CUDNN_CHECK(cudnnSetTensorNdDescriptor(hx_desc_, GetCudnnDataType(dtype), 3,
+                                         dimA, strideA));
+  CUDNN_CHECK(cudnnSetTensorNdDescriptor(cx_desc_, GetCudnnDataType(dtype), 3,
+                                         dimA, strideA));
+  CUDNN_CHECK(cudnnSetTensorNdDescriptor(hy_desc_, GetCudnnDataType(dtype), 3,
+                                         dimA, strideA));
+  CUDNN_CHECK(cudnnSetTensorNdDescriptor(cy_desc_, GetCudnnDataType(dtype), 3,
+                                         dimA, strideA));
+
+  size_t dropoutStatesSize;
+  CUDNN_CHECK(cudnnDropoutGetStatesSize(ctx->cudnn_handle, &dropoutStatesSize));
+  dropoutStates_ = Tensor(Shape{dropoutStatesSize}, dev, dtype);
+  CUDNN_CHECK(cudnnSetDropoutDescriptor(dropout_desc_, ctx->cudnn_handle, dropout_, this->dropoutStates_.block()->mutable_data(), dropoutStatesSize, 0x01234567));
+  
+  cudnnRNNInputMode_t inputMode;
+  cudnnDirectionMode_t direction;
+  cudnnRNNMode_t mode;
+  
+  if (inputMode_ == "cudnn_linear_input" || inputMode_ == "cudnn_skip_input"){
+    if (inputMode_ == "cudnn_linear_input")
+      inputMode = CUDNN_LINEAR_INPUT;
+    else if (inputMode_ == "cudnn_skip_input")
+      inputMode = CUDNN_SKIP_INPUT;
+  }
+  if (direction_ == "cudnn_undirectional" || direction_ == "cudnn_bidirectional"){
+    if (direction_ == "cudnn_undirectional")
+      direction = CUDNN_UNIDIRECTIONAL;
+    else if (direction_ == "cudnn_bidirectional")
+      direction = CUDNN_BIDIRECTIONAL;
+  }
+  if (mode_ == "cudnn_rnn_relu" || mode_ == "cudnn_rnn_tanh" ||
+        mode_ == "cudnn_lstm" || mode_ == "cudnn_gru"){
+    if (mode_ == "cudnn_rnn_relu")
+      mode = CUDNN_RNN_RELU;
+    else if (mode_ == "cudnn_rnn_tanh")
+      mode = CUDNN_RNN_TANH;
+    else if (mode_ == "cudnn_lstm")
+      mode = CUDNN_LSTM;
+    else if (mode_ == "cudnn_gru")
+      mode = CUDNN_GRU;
+  }
+  CUDNN_CHECK(cudnnSetRNNDescriptor(rnn_desc_, hiddenSize_, numLayers_, dropout_desc_, inputMode, direction, mode, GetCudnnDataType(dtype)));
+
+  // CUDNN_CHECK(cudnnGetRNNParamsSize(ctx->cudnn_handle, rnn_desc_, x_descs_[0], &weightSize_, GetCudnnDataType(dtype)));
```

Comment:
pls check the param size from cudnnGetRNNParamsSize and the one computed in Setup.